lib/oelite/profiling.py

```diff
@@ -0,0 +1,188 @@
+import datetime
+import functools
+import atexit
+import oelite.util
+import inspect
+import os
+import sys
+from resource import *
+
+now = datetime.datetime.utcnow
+
+profiledir = None
+
+# Decorating any function with @profile_calls will record the duration
+# of every call of that function. Some statistics on these are
+# automatically printed to $profiledir/callstats.txt on exit.
+profiled_functions = dict()
+def profile_calls(somefunc):
+    @functools.wraps(somefunc)
+    def recordtime(*args, **kwargs):
+        start = now()
+        try:
+            return somefunc(*args, **kwargs)
+        finally:
+            delta = (now()-start).total_seconds()
+            if not somefunc in profiled_functions:
+                profiled_functions[somefunc] = SimpleStats()
+            profiled_functions[somefunc].append(delta)
+    return recordtime
+
+def write_call_stats():
+    outfn = os.path.join(profiledir, "call_stats.txt")
```

Comment:
True, but (a) write_call_stats is only called as an atexit handler, which is only installed in init(), and (b) commit b4d235b (oelite/profiling.py: implement profile_output function) removes this problem altogether. I probably should squash that into the original commit creating profiling.py.